plugins/cleanupNumericValues.js

```diff
@@ -39,6 +39,7 @@ export const fn = (_root, params) => {
         if (node.attributes.viewBox != null) {
           const nums = node.attributes.viewBox.split(/\s,?\s*|,\s*/g);
           node.attributes.viewBox = nums
+            .filter((value) => value.length != 0)
```

Comment:
It would be easier and more performant to amend regex to something like `/\s+,?\s*|,\s*/g`

Reply:
Hmm, I don't see how that'd work.

For clarity, with your proposed regex, the split would be:

* `'0 0 100 100'` ⇒ `['0', ' ', '0', ' ', '100', ' ', '100']`
* `'0 0 100 100'` ⇒ `[' 0', ' ', '0', ' ', '100', ' ', '100 ']`
* `'0 0 100 100'` ⇒ `['  0', '  ', '0', '  ', '100', '  ', '100  ']`

I don't think this would be easy with regex alone, as it'd require a negative look behind with a non-fixed width, which isn't possible.

For example, to exclude the spaces at the end, `/(?:\s,?\s*|,\s*)(?!\s*$)/g` would work. But the same strategy can't be employed to exclude the spaces at the start.

Meanwhile, even if we were to ignore the different results, using either of the proposed regexes performs slower than a simple regex + iteration.

Something that actually would be both sensible and faster would be to `trim` the attribute instead of iterating after splitting, though.

i.e. `node.attributes.viewBox.trim().split(/\s,?\s*|,\s*/g)`

```js
const Benchmark = require('benchmark');
const suite = new Benchmark.Suite;

const data = [
  '0 0 100 100',
  ' 0 0 100 100 ',
  '  0  0  100  100  ',
]

/**
 * @param {string} viewbox
 * @return {string[]}
 */
const splitV1 = (viewbox) => {
  return viewbox.split(/\s,?\s*|,\s*/g).filter(v => v.length != 0);
};

/**
 * @param {string} viewbox
 * @return {string[]}
 */
const splitV2 = (viewbox) => {
  return viewbox.trim().split(/\s,?\s*|,\s*/g);
};

suite
  .add('v1', () => data.map((d) => splitV1(d)))
  .add('v2', () => data.map((d) => splitV2(d)))
  .on('cycle', (event) => console.log(String(event.target)))
  .run({ 'async': true });

// v1 x 2,648,671 ops/sec ±0.87% (91 runs sampled)
// v2 x 2,800,976 ops/sec ±0.53% (93 runs sampled)
```

Reply:
Number is ok to extra spaces, so all we need is to split numbers. The key factor here is to get rid of empty strings at the start and the end. So, to make it work in the middle, I've added `\b` separator (lookbehind and lookahead also would work, but `\b` is simpler). I forgot, that `split()` captures group, so non-capturing group is needed. So, the regex would be like: `/\b(?:\s+,?\s*|\s*,\s*)\b/g`.

Also, `matchAll()` can be used just to get numbers, but it will not be spec compliant — any separators would work.

Reply:
Ahh, yeah, that makes sense. And indeed, your proposal works now!

As we have existing test cases that support `-`, we should add that too.  
So `/\b(?:\s+,?\s*|\s*,\s*)(?=[-.\d])/g`, which includes a `-` in the look ahead.

Alternatively, we could also just use `[^\s]` to exclude spaces rather than a list of items to include, which simplifies the regex a little while offering more safety in case we missed something.  
So `/\b(?:\s+,?|\s*,)\s*(?=[^\s])/g`, this also moves the duplicate matcher for `\s*` to outside of the non-capture group.

```js
const data = [
  '0 0 100 100',
  ' 0 0 100 100 ',
  '  0  0  100  100  ',
  ' 0  0  0.5  .5 ',
  '20.000001 -19.99999 17.123456 70.708090',
];

const splitV1 = (viewbox) => {
  return viewbox.split(/\s,?\s*|,\s*/g).filter(v => v.length != 0);
};
// v1 x 1,591,214 ops/sec ±1.20% (92 runs sampled)

const splitV2 = (viewbox) => {
  return viewbox.trim().split(/\s,?\s*|,\s*/g);
};
// v2 x 1,585,072 ops/sec ±1.14% (92 runs sampled)

const splitV3 = (viewbox) => {
  return viewbox.split(/\b(?:\s+,?\s*|\s*,\s*)(?=[-.\d])/g);
};
// v3 x 1,684,750 ops/sec ±0.76% (94 runs sampled)

const splitV4 = (viewbox) => {
  return viewbox.split(/\b(?:\s+,?\s*|\s*,\s*)(?=[^\s])/g);
};
// v4 x 1,698,961 ops/sec ±0.68% (94 runs sampled)

const splitV5 = (viewbox) => {
  return viewbox.split(/\b(?:\s+,?|\s*,)\s*(?=[^\s])/g);
};
// v5 x 1,736,749 ops/sec ±0.43% (93 runs sampled)
```

Reply:
`[^\s]` is just `\S`

Reply:
I'd opened a pull request to apply the regex change, but closed it.

I've been getting more into benchmarking, and decided to finally revamp how I benchmark locally, which includes defining more rules and utilities.

One rule, is that functions must have identical input/output. Not "effectively identical", which is what I did before. This enforces a more complete benchmark, rather than only testing intermediate steps in an algorithm, where the performance of later steps may be impacted by the results of earlier steps.

For example, I treated `"100"` and `"  100"` as the same, because we know that both become `100` when called on `Number()`. However, this then doesn't account for the performance of `Number()` itself with the different inputs.

Writing a more complete benchmark has a notable impact on results:

```js
const data = [
  '0 0 120 120',
  '0 0 230 120',
  '110 0 120 120',
  '0 0 200.28423 200.28423',
  '20.000001 -19.99999 17.123456 70.708090',
  ' 0 0      150 100 ',
  '  0  0  0.5  .5  ',
];

function splitV2(viewbox) {
  return viewbox
    .trim()
    .split(/(?:\s,?|,)\s*/g)
    .map(Number);
};
// v2 x 619,413 ops/sec ±0.42% (97 runs sampled)

function splitV5(viewbox) {
  return viewbox
    .split(/\b(?:\s+,?|\s*,)\s*(?=\S)/g)
    .map(Number);
};
// v5 x 599,606 ops/sec ±0.42% (94 runs sampled)
```

The difference is minor (4-5%), but I'd argue using `trim` is simpler to maintain over a more complicated regex (I say this as someone that loves regex.) while offering on-par or better performance over the inputs we'd realistically encounter.

Reply:
But now we have `.filter()` instead of `.trim()`, which probably is worse.

Reply:
Yeah, `.filter()` is worse! I still plan to do a PR for this to use `trim` and to move the `\s*` outside the group. I was just dropping an update already with the findings.

Before opening another PR, just wanted to finish evaluating how I benchmark locally so I can handle this better in future, and review other decisions I'd make historically based on benchmarks.